### PR TITLE
fix: sort the events in each day cell by time

### PR DESCRIPTION
**Pull request by richardshiue**
### Feature Preview

fix: #2404

---

#### PR Checklist

- [x] My code adheres to the [AppFlowy Style Guide](https://appflowy.gitbook.io/docs/essential-documentation/contribute-to-appflowy/software-contributions/submitting-code/style-guides)
- [x] I've listed at least one issue that this PR fixes in the description above.
- [ ] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [ ] All existing tests are passing.

### CODE REVIEW

File: frontend/appflowy_flutter/lib/plugins/database_view/calendar/presentation/calendar_day.dart [hunk 1]
```diff
@@ -98,6 +98,9 @@ class CalendarDayCard extends StatelessWidget {
   }
 
   List<GestureDetector> _buildCards(BuildContext context) {
+    events.sort(
```
Comment (a-wallen) [inline frontend/appflowy_flutter/lib/plugins/database_view/calendar/presentation/calendar_day.dart:101]: This sorts the events every time the build method is called, no? If you convert this to a stateful widget, then you can sort the events in `initState()`.

Comment (a-wallen) [reply frontend/appflowy_flutter/lib/plugins/database_view/calendar/presentation/calendar_day.dart:101]: This was just a fly-by comment. I realized that converting to a stateful widget might not be the most appropriate solution. My only concern is that sorting can get expensive if we sort the events on every build.

Comment (appflowy) [reply frontend/appflowy_flutter/lib/plugins/database_view/calendar/presentation/calendar_day.dart]: I moved it to upper level that construct this widget.

**Comment by appflowy** [comment]
Build the card for each event  instead of building all of them.
image

**Pull request merged**